metrics_api/lib/opentelemetry/metrics/meter_provider.rb

```diff
@@ -10,11 +10,11 @@ module Metrics
     class MeterProvider
       # Returns a {Meter} instance.
       #
-      # @param [optional String] name Instrumentation package name
+      # @param [String] name Instrumentation package name
       # @param [optional String] version Instrumentation package version
       #
       # @return [Meter]
-      def meter(name = nil, version = nil)
+      def meter(name, version = nil)
         @meter ||= Meter.new
       end
```

Comment:
Also, we could take the opportunity to turn `version` into a keyword arg if that's the pattern we would like to have like suggested here: https://github.com/open-telemetry/opentelemetry-ruby/pull/1237#discussion_r878525464

```rb
  def meter(name, version: nil)
```

And in the future,

```rb
  def meter(name, version: nil, schema_url: nil, attributes: {})
```

Reply:
I think named arguments makes sense. There's some inconsistency between this and the Tracing API because the Metrics API explicitly requires a `name` whereas the Tracing API waffled a little on whether it was required, whether it was honoured (an implementation can ignore it), and what to do when an invalid value was passed. IMO we should move Metrics in our preferred direction and subsequently update the Tracing interface to match.

Reply:
Updated in https://github.com/open-telemetry/opentelemetry-ruby/pull/1441/commits/1c54ec3d1715e9944f2aa7b92f7b0b16da8528a1